multiversx_sdk_cli/cli_deps.py

```diff
@@ -34,16 +36,63 @@ def install(args: Any):
 
 def check(args: Any):
     name: str = args.name
-    module = dependencies.get_module_by_key(name)
-    tag_to_check: str = config.get_dependency_tag(module.key)
+
+    if name == "all":
+        all_dependencies = dependencies.get_all_deps()
+        missing_dependencies: List[Tuple[str, str]] = []
+
+        for dependency in all_dependencies:
+            tag_to_check: str = config.get_dependency_tag(dependency.key)
+            is_installed = check_module_is_installed(dependency, tag_to_check)
+
+            if not is_installed:
+                missing_dependencies.append((dependency.key, tag_to_check))
+
+        if len(missing_dependencies):
+            raise errors.DependenciesMissing(missing_dependencies)
+        return
+    if name == "rust":
+        module = dependencies.get_module_by_key(name)
+        tag_to_check: str = config.get_dependency_tag(module.key)
+
+        is_installed = check_module_is_installed(module, tag_to_check)
+        if is_installed:
+            actual_rust = _get_actual_installed_rust_version()
+
+            if tag_to_check in actual_rust:
+                logger.info(f"[{module.key} {tag_to_check}] is installed.")
+                return
+            if "command not found" in actual_rust:
+                logger.warning("You have installed Rust without using `rustup`.")
+                return
+            else:
+                logger.warning(f"The Rust version you have installed does not match the recommended version.\nInstalled [{actual_rust}], expected [{tag_to_check}].")
+                return
+
+        raise errors.DependencyMissing(module.key, tag_to_check)
+    else:
+        module = dependencies.get_module_by_key(name)
+        tag_to_check: str = config.get_dependency_tag(module.key)
+
+        is_installed = check_module_is_installed(module, tag_to_check)
+        if is_installed:
+            logger.info(f"[{module.key} {tag_to_check}] is installed.")
+            return
+
+        raise errors.DependencyMissing(module.key, tag_to_check)
+
+
+def check_module_is_installed(module: DependencyModule, tag_to_check: str) -> bool:
     resolution: str = config.get_dependency_resolution(module.key)
     resolution = resolution if resolution else "HOST"
 
     logger.info(f"Checking dependency: module = {module.key}, tag = {tag_to_check}, resolution = {resolution}")
 
     installed = module.is_installed(tag_to_check)
-    if installed:
-        logger.info(f"[{name} {tag_to_check}] is installed.")
-        return
+    return installed
+
 
-    raise errors.DependencyMissing(name, tag_to_check)
+def _get_actual_installed_rust_version() -> str:
```

Comment:
:+1:

Reply:
Can we have it directly in `Rust` dependency module - `modules.py`, by overriding `is_installed()`?

Reply:
Moved it in the `Rust` module.